### ISSUE TITLE
Exception in PropertyHasChanged()

### ISSUE THREAD

**Issue by PatrikTrestik** [mfidemraizer/trackerdog#44]
I'm getting 

> ArgumentException: Key was already added
> in 
> System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
>    v System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
>    v TrackerDog.ObjectChangeTrackingExtensions.GetPropertyTracking(Object some, String propertyName)

target.GetChangeTracker().ChangedProperties shows correctly changed property info.

I'm posting here to check if I'm doing something wrong.
Example and Source debugging will follow.

**Comment by mfidemraizer** [comment]
Yes, please provide an example to further investigate the issue! :+1:

**Comment by PatrikTrestik** [comment]
Hi. I had to change for a different solution because of the very high performance impact of TrackingDog. It slowed down reading of 800 records 10 times.

I'll try to investigate the above error further but I have to do it in my spare time as my work project no more utilizes TrackingDog. It can take some time, sorry.

**Comment by mfidemraizer** [comment]
@PatrikTrestik  Does your workplace's project use an OR/M already? Anyway, creating proxies has some perfomance impact. 

I doubt, anyway, that TrackerDog could slowdown 10 times the perf: it should be something with how things were configured.

**Comment by mfidemraizer** [comment]
@PatrikTrestik Also, the way you use TrackerDog may impact the perfomance too. I should know the use case...
